src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -704,9 +718,59 @@ public void DeleteRevisionsFor(DocumentsOperationContext context, string id, lon
                 var newEtag = _documentsStorage.GenerateNextEtag();
                 var changeVector = _documentsStorage.GetNewChangeVector(context, newEtag);
                 context.LastDatabaseChangeVector = changeVector;
+
+                var lastModifiedTicks = _database.Time.GetUtcNow().Ticks;
+                var deletedRevisionsCount = 0L;
+                if (onlyForceCreated)
+                    deletedRevisionsCount = DeleteForceCreatedRevisions(context, table, prefixSlice, collectionName, changeVector, lastModifiedTicks);
+                else
+                    deletedRevisionsCount = DeleteRevisions(context, table, prefixSlice, collectionName, numberOfRevisionsToDelete: long.MaxValue, minimumTimeToKeep: null, changeVector, lastModifiedTicks);
```

Comment:
We want to delete revisions for a given document, by default we will not touch the `ForceCreated` ones and remove the normal revisions.
if `deleteForceCreated` is set to `true`, we will remove them all.

Reply:
Done